error[E0277]: the trait bound `St: BuildHasher` is not satisfied

error[E0277]: the trait bound `St: BuildHasher` is not satisfied

```bash
  --> /var/home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/counter-0.7.0/src/impls/serialize.rs:31:20
   |
31 |         let map = <_>::deserialize(deserializer)?;
   |                    ^ the trait `BuildHasher` is not implemented for `St`
   |
   = note: required for `HashMap<T, N, St>` to implement `Deserialize<'_>`
help: consider further restricting type parameter `St` with trait `BuildHasher`
   |
25 |     St: Default + std::hash::BuildHasher,
   |                 ++++++++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `counter` (lib) due to 1 previous error
```

My code:
```rust

    let words_frequency = txt_file.split_whitespace().collect::<Counter<_>>() // text_file: String

    let mut words_frequency_file = OpenOptions::new()
        .create(true)
        .truncate(true) // If the file already exists we want to overwrite the old data
        .write(true)
        .read(true)
        .open("words_frequency.yaml").unwrap();
    yaml_serde::to_writer(words_frequency_file, &words_frequency);
```

I could try to create merge request, but I'm learning rust so probably only try

--- a/src/impls/serialize.rs
+++ b/src/impls/serialize.rs
@@ -1,4 +1,4 @@
-use std::hash::Hash;
+use std::hash::{Hash, BuildHasher};
 
 use num_traits::Zero;
 use serde::{de::Deserializer, ser::Serializer, Deserialize, Serialize};
@@ -22,7 +22,7 @@ impl<'de, T, N, St> Deserialize<'de> for Counter<T, N, St>
 where
     T: Deserialize<'de> + Hash + Eq,
     N: Deserialize<'de> + Zero,
-    St: Default,
+    St: Default + BuildHasher,
 {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where

